src/plans/index.ts

```diff
@@ -65,15 +78,46 @@ export type PlanVersions = {
      */
     getCurrencies: (planUuid: string) => Promise<PlanCurrency[]>;
   };
-};
+  [Version.V3]: {
+    /**
+     *  Get all plans
+     *
+     * @param GetAllPlansOptionsV3
+     *
+     *  @returns { Promise<GetAllFeaturesV3>}
+     */
+    getAll: (options?: GetAllPlansOptionsV3) => Promise<GetAllPlansV3>;
+    /**
+     *  Retrieves all plans for an organisation with cursor based pagination. The response does not contain any relational data. If you want to expand the relational data, you can do so with the `expand` parameter.
+     **
+     * Docs - https://docs.salable.app/api/v3#tag/Plans/operation/getPlanByUuid
+     *
+     * @returns {Promise<PlanV3 & {
+     *       features?: PlanFeatureV3[];
+     *       currencies?: PlanCurrency[];
+     *       product?: ProductV3 & { organisationPaymentIntegration: OrganisationPaymentIntegrationV3 }
+     *     }>}
+     */
+    getOne: (
+      planUuid: string,
+      options?: GetPlanOptionsV3
+    ) => Promise<GetAllPlansV3>;
 
-export type PlanVersionedMethods<V extends TVersion> = V extends keyof PlanVersions ? PlanVersions[V] : never;
+    /**
 
-export const plansInit = <V extends TVersion>(version: V, request: ApiRequest): PlanVersionedMethods<V> => {
-  switch (version) {
-    case Version.V2:
-      return v2PlanMethods(request) as PlanVersionedMethods<V>;
-    default:
-      throw new Error('Unsupported version');
-  }
+     /**
+     * Retrieves a checkout link for a specific plan. The checkout link can be used by customers to purchase the plan.
+     *
+     * @param  {string} planUuid The UUID of the plan
```

Comment:
Should there be a `-` here like options has?


```suggestion
     * @param  {string} planUuid - The UUID of the plan
```